test/functional/apps/discover/_indexpattern_with_unmapped_fields.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const kibanaServer = getService('kibanaServer');
+  const testSubjects = getService('testSubjects');
+  const log = getService('log');
+  const retry = getService('retry');
+  const PageObjects = getPageObjects(['common', 'timePicker', 'discover']);
+
+  describe('index pattern with unmapped fields', () => {
+    const unmappedFieldsSwitchSelector = 'unmappedFieldsSwitch';
+
+    before(async () => {
+      await esArchiver.loadIfNeeded('unmapped_fields');
+      await kibanaServer.uiSettings.replace({ defaultIndex: 'test-index-unmapped-fields' });
+      await kibanaServer.uiSettings.update({
+        'discover:searchFieldsFromSource': false,
+      });
```

Comment:
I think you can merge those modification of uiSettings

Reply:
It didn't work the unless explicitly updating